st2common/bin/st2-self-check

```diff
@@ -159,7 +159,7 @@ if [ ${RUN_ORQUESTA_TESTS} = "true" ]; then
 
     START_TS=$(date +%s)
     OUTPUT=$(st2 run examples.orquesta-examples)
-    echo ${OUTPUT} | grep "status" | grep -q "succeeded"
+    echo ${OUTPUT} | grep -q "status: succeeded"
```

Comment:
`^` should be used to avoid matching `status: succeeded` in the `log` section.

Reply:
OK, I see now that it's indeed (!) matches the status from the log section :) 
It happens because of the colors in `st2 run` output. I can understand now why the previous check was hacky `grep "status" | grep -q "succeeded"`.

`status: succeeded` in the log section works well too, as it refers to the parent workflow run.

Reply:
1. St2 doesn't appear to provide a `--nocolour` option :disappointed: 
2. A workaround would be to output in json format and extract the value using `jq`, unfortunately, json output is broken
```# st2 execution get -j 61b1daecbf16900d8407e314 | jq -s .[].status
"succeeded"
jq: error (at <stdin>:100): Cannot index array with string "status"
```
Which looks to be related to this issue https://github.com/StackStorm/st2/issues/4639

I also tried `--raw`, `-k status` and a few other options, all of them appear broken

Reply:
With that, I think what's there with grep is sufficient to check if the parent workflow was good :heavy_check_mark:

Reply:
```suggestion
    echo ${OUTPUT} | grep -q "^status: succeeded"
```

Reply:
@cognifloyd That won't work. See https://github.com/StackStorm/st2/pull/5487#discussion_r765882873

Reply:
@armab  Is the color only around the status value, so can we use "^status" and then a grep on succeeded - so it only matches the grep on the status and not log.
Or can we instead grep on "succeeded: true" perhaps from the result section? Or "failed: false" from the result section?

Reply:
```suggestion
    echo ${OUTPUT} | grep -q "^[^ ]*status[^ ]*: [^ ]*succeeded"
```
Then match anything except a space?

I'm on my phone so I can't test atm.

Reply:
@amanda11 Just tried that. Interesting, but looks like `result.failed` and `result.succeeded` might be missing in the output depending on the workflow result or type.


Example: 
```
Attempting Test tests.test_inquiry_chain...ERROR! (51s)
Test output: .........................
id: 61b2624549bdb8b48ff0c53b
action.ref: tests.test_inquiry_chain
parameters: 
  protocol: http
  token: 8fa1b7c690084a2ca756deb857fae14e
status: succeeded
result_task: assert_workflow_expected_output
result: 
  exit_code: 0
  result: true
  stderr: ''
  stdout: EQUAL
start_timestamp: Thu, 09 Dec 2021 20:08:37 UTC
end_timestamp: Thu, 09 Dec 2021 20:09:26 UTC
log: 
  - status: requested
    timestamp: '2021-12-09T20:08:37.933000Z'
  - status: scheduled
    timestamp: '2021-12-09T20:08:38.004000Z'
  - status: running
    timestamp: '2021-12-09T20:08:38.198000Z'
  - status: succeeded
    timestamp: '2021-12-09T20:09:26.835000Z'
+--------------------------+------------------------+----------------------------------+----------------------------------+-------------------------------+
| id                       | status                 | task                             | action                           | start_timestamp               |
+--------------------------+------------------------+----------------------------------+----------------------------------+-------------------------------+
| 61b26246dd2d33e408d589a8 | succeeded (3s elapsed) | execute_inquiry_workflow         | core.local                       | Thu, 09 Dec 2021 20:08:38 UTC |
| 61b2624add2d33e408d589aa | succeeded (1s elapsed) | get_inquiry_trigger              | core.local                       | Thu, 09 Dec 2021 20:08:42 UTC |
| 61b2624bdd2d33e408d589ac | succeeded (1s elapsed) | get_inquiry_id                   | core.local                       | Thu, 09 Dec 2021 20:08:43 UTC |
| 61b2624cdd2d33e408d589ae | succeeded (1s elapsed) | get_workflow_id                  | core.local                       | Thu, 09 Dec 2021 20:08:44 UTC |
| 61b2624ddd2d33e408d589b0 | succeeded (2s elapsed) | get_workflow_details_1           | core.local                       | Thu, 09 Dec 2021 20:08:45 UTC |
| 61b2624fdd2d33e408d589b2 | failed (4s elapsed)    | invalid_response_expect_failure  | core.local                       | Thu, 09 Dec 2021 20:08:47 UTC |
| 61b26254dd2d33e408d589b4 | succeeded (6s elapsed) | pause_after_invalid_response     | core.pause                       | Thu, 09 Dec 2021 20:08:52 UTC |
| 61b2625bdd2d33e408d589b6 | succeeded (1s elapsed) | get_workflow_details_2           | core.local                       | Thu, 09 Dec 2021 20:08:59 UTC |
| 61b2625ddd2d33e408d589f1 | succeeded (4s elapsed) | valid_response                   | core.local                       | Thu, 09 Dec 2021 20:09:01 UTC |
| 61b26261dd2d33e408d589f3 | succeeded (5s elapsed) | pause_after_valid_response       | core.pause                       | Thu, 09 Dec 2021 20:09:05 UTC |
| 61b26267dd2d33e408d589f5 | succeeded (1s elapsed) | get_workflow_details_3           | core.local                       | Thu, 09 Dec 2021 20:09:11 UTC |
| 61b26268dd2d33e408d589f7 | failed (4s elapsed)    | duplicate_response_expect_failur | core.local                       | Thu, 09 Dec 2021 20:09:12 UTC |
|                          |                        | e                                |                                  |                               |
| 61b2626cdd2d33e408d589f9 | succeeded (1s elapsed) | assert_triggers                  | asserts.object_key_string_equals | Thu, 09 Dec 2021 20:09:16 UTC |
| 61b2626ddd2d33e408d589fb | succeeded (1s elapsed) | assert_workflow_paused_prep      | core.local                       | Thu, 09 Dec 2021 20:09:17 UTC |
| 61b2626fdd2d33e408d589fd | succeeded (0s elapsed) | assert_workflow_paused           | asserts.object_equals            | Thu, 09 Dec 2021 20:09:19 UTC |
| 61b26270dd2d33e408d589ff | succeeded (0s elapsed) | assert_workflow_still_paused_pre | core.local                       | Thu, 09 Dec 2021 20:09:20 UTC |
|                          |                        | p                                |                                  |                               |
| 61b26271dd2d33e408d58a01 | succeeded (0s elapsed) | assert_workflow_still_paused     | asserts.object_equals            | Thu, 09 Dec 2021 20:09:21 UTC |
| 61b26272dd2d33e408d58a04 | succeeded (0s elapsed) | assert_workflow_succeeded_prep   | core.local                       | Thu, 09 Dec 2021 20:09:22 UTC |
| 61b26273dd2d33e408d58a06 | succeeded (0s elapsed) | assert_workflow_succeeded        | asserts.object_equals            | Thu, 09 Dec 2021 20:09:23 UTC |
| 61b26274dd2d33e408d58a08 | succeeded (0s elapsed) | assert_workflow_expected_output_ | core.local                       | Thu, 09 Dec 2021 20:09:24 UTC |
|                          |                        | prep                             |                                  |                               |
| 61b26275dd2d33e408d58a0a | succeeded (1s elapsed) | assert_workflow_expected_output  | asserts.object_key_string_equals | Thu, 09 Dec 2021 20:09:25 UTC |
+--------------------------+------------------------+----------------------------------+----------------------------------+-------------------------------+

```

We're a bunch of bros :man_facepalming:

Reply:
See https://github.com/StackStorm/st2/pull/5487#pullrequestreview-828201395 for the alternative fix.